backend/src/HueIT-backend.kt

```diff
@@ -19,66 +22,84 @@ import io.ktor.routing.post
 import io.ktor.routing.routing
 import io.ktor.server.engine.embeddedServer
 import io.ktor.server.netty.Netty
-import org.json.simple.JSONObject
 import org.slf4j.event.Level
 import java.util.*
 import kotlin.collections.ArrayList
 import kotlin.collections.set
 
 val client = HttpClient()
-val debug = false
+val mapper = ObjectMapper()
+const val hueConst: Double = 65535.0 / 360
+const val satConst: Double = 254.0 / 100
+const val briConst: Double = 254.0 / 100
 
+@JsonInclude(JsonInclude.Include.NON_NULL)
 data class ResponseList(
-        val responses: List<ResponseBody>,
-        val errors: List<ErrorBody>?
+    val responses: List<ResponseBody>,
+    val errors: List<ErrorBody>?
 )
 
+@JsonInclude(JsonInclude.Include.NON_NULL)
 data class ResponseBody(
-        val isGroup: Boolean,
-        val id: Int,
-        val updatedProps: RequestBodyProperty?
+    val isGroup: Boolean,
+    val id: Int,
+    val updatedProps: RequestBodyProperty?
 )
 
 data class ErrorBody(
-        val type: Int,
-        val isGroup: Boolean,
-        val id: Int,
-        val description: String
+    val type: Int,
+    val isGroup: Boolean,
+    val id: Int,
+    val description: String
 )
 
 data class RequestBodyList(
-        val requestBodyList: List<RequestBody>
+    val requestBodyList: List<RequestBody>
 )
 
+@JsonInclude(JsonInclude.Include.NON_NULL)
 data class RequestBody(
-        val isGroup: Boolean,
-        val id: Int,
-        val props: RequestBodyProperty?
+    val isGroup: Boolean,
+    val id: Int,
+    val props: RequestBodyProperty?
 )
 
+@JsonInclude(JsonInclude.Include.NON_NULL)
 data class RequestBodyProperty(
-        val pwr: Boolean?,
-        val hue: Double?,
-        val sat: Double?,
-        val bri: Double?,
-        val rst: Boolean?
+    val pwr: Boolean?,
```

Comment:
Please don't use abbreviations. Does this mean Power? Password? Something else?

Reply:
This has been fixed in a recent commit now.